rust/worker/src/execution/operators/merge_metadata_results.rs

```diff
@@ -35,6 +35,8 @@ pub struct MergeMetadataResultsOperatorInput {
     filtered_offset_ids: Option<Vec<u32>>,
     record_segment_definition: Segment,
     blockfile_provider: BlockfileProvider,
+    offset: Option<u32>,
```

Comment:
I'd prefer if this was not an option at this layer, and that default behavior was handled by the caller.